shuffle-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/ucx/UCX.scala

```diff
@@ -117,6 +120,27 @@ class UCX(executorId: Int, usingWakeupFeature: Boolean = true) extends AutoClose
   // the worker thread. We need this to complete prior to getting the `rkeys`.
   private var pendingRegistration = false
 
+  // Error handler that would be invoked on endpoint failure.
+  private val epErrorHandler = new UcpEndpointErrorHandler {
+    override def onError(ucpEndpoint: UcpEndpoint, errorCode: Int, errorString: String): Unit = {
+      endpoints.values().removeIf(ep => ep == ucpEndpoint)
+      ucpEndpoint.close()
```

Comment:
correct me if I am wrong, could this `ucpEndpoint` be one of the ones created regularly without the listener (i.e. what we do today?)

And then when `.close` happens, what happens to requests that are outstanding?

Since this is being called from the progress thread, what will happen (I think) is something like this:

`[good work for epId1, ..., epId1 error detected (callback called), ..., more work for epId1 that will now fail]`

For the last work that can't find the missing `epId`, I think we'll get a bunch of exceptions when it can't find the endpointId that was removed, this is happening while we drain the queue of work for the ep that was missing. That said, the code isn't calling the error callback, so it's going to leave the application in a bad state.

Since this is all behind an if statement, it is fine for this PR. I just need to make sure the follow ups are documented and we get working on them soon here.

Reply:
And then when .close happens, what happens to requests that are outstanding?

Send/flush/am requests would completed with `onError` callback and error code something like `UCS_ERR_IO_ERROR` or `UCS_ERR_NOT_CONNECTED` etc.

Reply:
https://github.com/NVIDIA/spark-rapids/issues/2275